### PR TITLE
RotateContext handle command tests

### PR DESCRIPTION
**Pull request by hpya93**
Hi @jhand2
Added tests for RotateContextHandle command
- Added a test to check and report invalid handle with a test handle  (moved to negativeCases.go, taken care in PR 255, as a common negative case test)

The following are part of this PR:
- RotateHandle in default context  
- RotateHandle in non-default context
- RotateHandle in with TARGET_DEFAULT flag

### CODE REVIEW

File: verification/verification.go [hunk 1]
```diff
@@ -53,12 +59,34 @@ var UnsupportedCommand = TestCase{
...
 	CertifyKeySimulationTestCase,
 	GetCertificateChainTestCase,
 	TagTCITestCase,
+	ExtendTCITestCase,
+	ExtendDerivedTciTestCase,
+	SignAsymmetricTestCase,
+	SignSymmetricTestCase,
+	SignSimulationTestCase,
+	RotateContextTestCase,
+	RotateContextSimulationTestCase,
```
Comment (hpya93) [inline verification/verification.go:89]: Tests RotateContext command in Default context and Simulation context.

File: verification/verification.go [hunk 2]
```diff
@@ -53,12 +59,34 @@ var UnsupportedCommand = TestCase{
...
+}
+var SignSimulationTestCase = TestCase{
+	"SignSimulation", TestSignSimulation, []string{"Simulation"},
+}
+var RotateContextTestCase = TestCase{
+	"RotateContextHandle", TestRotateContextHandle, []string{"AutoInit", "RotateContext"},
+}
+var RotateContextSimulationTestCase = TestCase{
+	"RotateContextHandleSimulation", TestRotateContextHandleSimulation, []string{"Simulation", "RotateContext"},
+}
```
Comment (hpya93) [inline verification/verification.go:76]: Added test cases and required DPE support for RotateContext command

File: verification/rotateContextHandle.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+func TestRotateContextHandle(d TestDPEInstance, c DPEClient, t *testing.T) {
+	testRotateContextHandle(d, c, t, false)
+}
+
+func TestRotateContextHandleSimulation(d TestDPEInstance, c DPEClient, t *testing.T) {
+	testRotateContextHandle(d, c, t, true)
+}
+
+func testRotateContextHandle(d TestDPEInstance, c DPEClient, t *testing.T, simulation bool) {
+	handle := getInitialContextHandle(d, c, t, simulation)
```
Comment (hpya93) [inline verification/rotateContextHandle.go:19]: Tests the RotateContext command with command flag.
The spec provides only one command flag TARGET_IS_DEFAULT.
Have checked used/unused command flag in default/simulation mode

Comment (hpya93) [reply verification/rotateContextHandle.go:19]: Hi @jhand2 
Kindly take a look at this PR containing tests for ROtateContext() command
This may be considered for review once PRs 253 and 251 are merged to overlook changes pertaining to ExtendTCi and  Sign command.

**Comment by hpya93** [comment]
Synced branch

File: verification/verification.go [hunk 4]
```diff
@@ -56,13 +56,21 @@ var UnsupportedCommand = TestCase{
 var UnsupportedCommandFlag = TestCase{
 	"CheckSupportForCommmandFlag", TestUnsupportedCommandFlag, []string{"AutoInit", "RotateContext", "ExtendTci"},
 }
+var RotateContextTestCase = TestCase{
+	"RotateContextHandle", TestRotateContextHandle, []string{"AutoInit", "RotateContext"},
+}
+var RotateContextSimulationTestCase = TestCase{
+	"RotateContextHandleSimulation", TestRotateContextHandleSimulation, []string{"Simulation", "RotateContext"},
+}
```
Comment (hpya93) [inline verification/verification.go:64]: Added test cases and required DPE support for RotateContext command

File: verification/rotateContextHandle.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+	return rotatedContext
+}
+
+// Checks whether the context handle is rotated to default context.
+// In simulated context, the rotated handle cannot be made default context handle as a mixture of default and
+// non-default context handles are not allowed.
+// In default context, the rotated handle can be made default context handle when there is no default context handle already.
+func testWithTargetDefaultFlag(d TestDPEInstance, c DPEClient, t *testing.T, handle *ContextHandle, simulation bool) {
+	if simulation {
```
Comment (hpya93) [inline verification/rotateContextHandle.go:52]: With TARGET_DEFAULT flag set, it is expected that rotated context handle must be a default context handle

**Comment by hpya93** [comment]
@jhand2 conflicts are resolved for this PR, could this be reviewed and merged?

File: verification/rotateContextHandle.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+		_, err := c.RotateContextHandle(handle, TargetIsDefault)
+		if err == nil {
+			t.Fatalf("[FATAL]: Should return %q for simulation context, but returned no error", StatusInvalidArgument)
+		} else if !errors.Is(err, StatusInvalidArgument) {
+			t.Fatalf("[FATAL]: Incorrect error type. Should return %q, but returned %q", StatusInvalidArgument, err)
+		}
```
Comment (jhand2) [inline verification/rotateContextHandle.go:58]: This function is really testing separate things for simulation and non-simulation. I think this check (that a simulation context can't be made default) should just be moved to be a part of `TestRotateContextHandleSimulation`

File: verification/rotateContextHandle.go [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+	// Check handle rotated is non-default when no flags are set
+	handle = testWithoutTargetDefaultFlag(d, c, t, handle)
+
+	// Check handle is rotated to default handle in default context and error is reported in simulated context
+	testWithTargetDefaultFlag(d, c, t, handle, simulation)
+}
+
+// Checks whether the default context handle is rotated to a some other context handle
+func testWithoutTargetDefaultFlag(d TestDPEInstance, c DPEClient, t *testing.T, handle *ContextHandle) *ContextHandle {
```
Comment (jhand2) [inline verification/rotateContextHandle.go:34]: This function has a bit of a side-effect where it rotates the default context. Subsequent tests only succeed because `testWithTargetDefaultFlag` is called after.

I think we should split out the logic for simulation and non-simulation because the tests are checking different things.

For the non-simulation test, we have get the default context, rotate it to be non-default, and then rotate it back. I think the fact that these operations are done in two separate helpers makes the code pretty hard to follow.

File: verification/rotateContextHandle.go (hunk 6 above)
Comment (hpya93) [reply verification/rotateContextHandle.go]: Yes, this is done, I have added the logic under TestRotateContextHandle() and TestRotateContextHandleSimulation()

File: verification/rotateContextHandle.go (hunk 7 above)
Comment (hpya93) [reply verification/rotateContextHandle.go]: Yes, this is done, I have added the logic under TestRotateContextHandle() and TestRotateContextHandleSimulation()
I earlier thought to have a function for each case like - what if the flag is set in Default/Sim mode, what if the flag is not set in Default/Sim mode. May that became the source of confusion. Anyway I fixed it now.

Comment (hpya93) [reply verification/rotateContextHandle.go]: @jhand2 , let me know if this is fine after addressing the review comments

**Review by jhand2 — APPROVED** [review]

File: verification/rotateContextHandle.go (hunk 7 above)
Comment (jhand2) [reply verification/rotateContextHandle.go]: Thanks! Looks good.
